AWSClientRuntime/Sources/Regions/IMDSRegionProvider.swift

```diff
@@ -0,0 +1,38 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+import AwsCommonRuntimeKit
+        
+public struct IMDSRegionProvider: RegionProvider {
+    private let REGION_PATH = "/latest/meta-data/placement/region"
+    let imdsClient: IMDSClient?
+    
+    public init() {
+        do {
+            self.imdsClient = try IMDSClient()
+        } catch {
+            self.imdsClient = nil
+        }
+    }
+    
+    public func resolveRegion() -> Future<String?> {
+        let future = Future<String?>()
+        guard let imdsClient = imdsClient else {
+            future.fulfill(nil)
```

Comment:
Add `log.err()` if we fail to instantiate IMDSClient()

Reply:
removed as we no longer have an optional client